qiskit/pulse/schedule.py

```diff
@@ -1021,3 +1001,26 @@ def _check_nonnegative_timeslot(timeslots):
                 raise PulseError(
                     "An instruction on {} has a negative "
                     " starting time.".format(chan))
+
+def _reduce_to_stack(schedule):
+    """Loops through pulse instructions and removes delays and zero phase shifts.
+    Args:
+        schedule: Schedule to insert.
+    Returns:
+        A tuple of pulse instructions with delays and zero phase shifts removed.
+    """
+
+    new_schedule = []
+    for inst in schedule.instructions:
+        time, action = inst
+        # Check if the action is a Delay()
+        if re.search(r'\bDelay\b', str(action)):
+            continue
```

Comment:
There is a method in python that should be used in this situation, called `isinstance`. This is important, because it is more robust than this method. Think of a scenario where we create a new instruction, `NewSuperDelay` that does something besides simply delaying and _does_ affect equality. In this case, your code would no longer work. Generally, if you can avoid using `re`, that is best.

```suggestion
        if isinstance(inst, Delay):
            continue
```
Notice as well that this implementation makes the comment redundant -- this indicates well written code that other developers will easily understand in the future (and thus code that is easier to maintain).

Same goes for `ShiftPhase` below.